user_guide_src/source/incoming/controllers.rst

```diff
@@ -109,28 +109,195 @@ then trying to access it using the following URL will not work::
 
     example.com/index.php/helloworld/utility/
 
+.. _controller-auto-routing-improved:
+
+Auto Routing (Improved)
+************************
+
+Since v4.2.0, the new more secure Auto Routing has been introduced.
+
+This section describes the functionality of the new auto-routing.
+It automatically routes an HTTP request, and executes the corresponding controller method
+without route definitions.
+
+Since v4.2.0, the auto-routing is disabled by default. To use it, see :ref:`enabled-auto-routing-improved`.
+
+Consider this URI::
+
+    example.com/index.php/helloworld/
+
+In the above example, CodeIgniter would attempt to find a controller named ``App\Controllers\Helloworld`` and load it, when auto-routing is enabled.
+
+.. note:: When a controller's short name matches the first segment of a URI, it will be loaded.
+
+Let's try it: Hello World!
+==========================
+
+Let's create a simple controller so you can see it in action. Using your text editor, create a file called **Helloworld.php**,
+and put the following code in it. You will notice that the ``Helloworld`` Controller is extending the ``BaseController``. you can
+also extend the ``CodeIgniter\Controller`` if you do not need the functionality of the BaseController.
+
+The BaseController provides a convenient place for loading components and performing functions that are needed by all your
+controllers. You can extend this class in any new controller.
+
+.. literalinclude:: controllers/020.php
+
+Then save the file to your **app/Controllers/** directory.
+
+.. important:: The file must be called **Helloworld.php**, with a capital ``H``. Controller class names MUST start with an uppercase letter and ONLY the first character can be uppercase.
```

Comment:
What do you mean by `explored this`?

This explanation (specification) came from the legacy version and CI3.

In fact, you could use `HelloWorld` when you navigate to `/helloWorld`.
Because the implementation does only `ucfirst()` to look for the Controller.
So if you have `HelloWorld` and navigate to `/helloworld`, it looks for `Helloworld` controller,
and it can't be found on case sensitive filesystem.

Reply:
Nope, this was just me being dull. 😓 I think yesterday I thought this was a change but obviously this is how it has been forever. I think I'm muddling Rails where controllers usually include the suffix: "UserController.rb" to CodeIgniter's "Users.php"